integration/test/Testlib/ModService.hs

```diff
@@ -365,22 +399,15 @@ logToConsole colorize prefix hdl = do
           `E.catch` (\(_ :: E.IOException) -> pure ())
   go
 
-waitUntilServiceUp :: HasCallStack => String -> Service -> App ()
-waitUntilServiceUp domain = \case
+retryRequestUntil :: HasCallStack => (Service -> App Bool) -> Service -> App ()
+retryRequestUntil mkReq = \case
```

Comment:
We could just accept an `App Bool` instead of `(Service -> App Bool)` and then also accepting a `Service` just to pass it to that function. This way we can achieve two things:
1. Rename the first `mkReq` function as `checkServiceUp`, so it is more clear what request it makes
2. We can avoid having to pass the awkward `FederatorInternal` as a service name when we're trying to see if the ingress is working fine.